lib/oxide-vpc/src/engine/overlay.rs

```diff
@@ -1021,35 +1025,12 @@ impl Default for Mcast2Phys {
     }
 }
 
-/// An overlay multicast group address mapped to the underlay (outer) IPv6 multicast address.
-///
-/// This type ensures that the wrapped IPv6 address is a valid multicast address
-/// with administrative scope (admin-local, site-local, or organization-local).
+/// Transparent wrapper for underlay IPv6 multicast addresses.
 ///
-/// Administrative scopes per RFC 4291 and RFC 7346:
-/// - `0x4`: admin-local scope
-/// - `0x5`: site-local scope
-/// - `0x8`: organization-local scope
+/// This newtype exists only to satisfy the orphan rule for implementing
+/// `ResourceEntry`. Validation is performed at the API boundary (xde.rs).
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
-pub struct MulticastUnderlay(Ipv6Addr);
-
-impl MulticastUnderlay {
-    /// Create a new `MulticastUnderlay` if the address is a valid
-    /// administratively-scoped multicast IPv6 address (scope 0x4, 0x5, or 0x8).
-    pub fn new(addr: Ipv6Addr) -> Option<Self> {
-        if addr.is_admin_scoped_multicast() { Some(Self(addr)) } else { None }
-    }
-
-    /// Return the underlying IPv6 multicast address.
-    pub fn addr(&self) -> Ipv6Addr {
-        self.0
-    }
-
-    /// Return the destination MAC address derived from the IPv6 multicast address.
-    fn dst_mac(&self) -> MacAddr {
-        self.0.unchecked_multicast_mac()
-    }
-}
+pub struct MulticastUnderlay(pub Ipv6Addr);
```

Comment:
I would generally shy away from having a newtype like this expose its innards as `pub`, since that makes it trivial to construct an invalid instance.

Reply:
definitely. meant to go back and clean this up.